js/repl/Repl.js

```diff
@@ -474,10 +504,57 @@ class Repl extends React.Component {
       .then(result => {
         result.meta.compiledSize = prettySize(result.meta.compiledSize);
         result.meta.rawSize = prettySize(result.meta.rawSize);
+        if (result.sourceMap) {
+          result.sourceMapRepresentation = this.processSourcemap(result.sourceMap);
+        }
         this.setState(result, setStateCallback);
       });
   };
 
+  processSourcemap = (rawSourceMap: string) => {
+    const smc = new sourceMap.SourceMapConsumer(rawSourceMap);
+
+    const mappings = [];
+    smc.eachMapping(function (m) {
+      if (m.name) mappings.push(m);
```

Comment:
This logic isn't quite what I'd recommend. If we're going to show mappings, we shouldn't exclude things, especially not based on if there is a name. Along with that:

* Mappings have both a start and an end
* Each original mappings can have multiple generated mappings.

Here is how I would read the mappings:

```
const originalPositions = [];
smc.eachMapping(
  function (m) {
    if (m.originalLine === null) return;

    const last = originalPositions.length > 0 ? originalPositions[originalPositions.length - 1] : null;

    // Ignore multiple matching original locations.
    if (last && last.line === m.originalLine && last.columnStart === m.originalColumn) return;

    // Set the _end_ of the previous mapping to the start of the new one.
    if (
      last &&
      last.line === m.originalLine
    ) {
      last.columnEnd = m.originalColumn;
    }

    originalPositions.push({
      source: m.source,
      name: m.name,
      line: m.originalLine,
      columnStart: m.originalColumn,
      columnEnd: Infinity,
    });
  },
  null,
  sourceMap.SourceMapConsumer.ORIGINAL_ORDER
);

// Compute the spans on the generated file.
smc.computeColumnSpans();

const results = originalPositions.forEach(origPos => {
  const ranges = smc.allGeneratedPositionsFor({
    source: origPos.source,
    line: origPos.line,
    column: origPos.columnStart,
  });

  const generatedRanges = ranges.map(range => ({
    line: range.line,
    columnStart: range.column,
    columnEnd: range. lastColumn + 1
  }));

  results.push({
    original: origPos,
    generated: generatedRanges,
  });
});
```
which will give you an array of the original ranges, and for each one of those, the _multiple_ generated ranges they reference.